pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java

```diff
@@ -520,13 +520,21 @@ public void destroy() {
         LOGGER.error("Failed to close inverted index. Service will continue with memory leaks, error: ", e);
       }
     }
+
+    for (Map.Entry<String, MutableDictionary> entry : dictionaryMap.entrySet()) {
```

Comment:
If close fails, I would like to log an error with the column name. So, it is better to get the entries rather than just values. entry.key() is the column name.